src/web/src/app/space/add/component.ts

```diff
@@ -0,0 +1,135 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { startWith, map, debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+import { MatOption } from '@angular/material/core';
+import { Router, NavigationExtras } from '@angular/router';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+
+import { SpaceService } from '../service';
+import { Space } from '../model';
+import { SpaceKeyValidation } from './space-key-validator';
+import { RequestWatcherService } from '../../request-watcher/service';
+
+@Component({
+	selector: 'app-space-add',
+	templateUrl: './component.html',
+	styleUrls: ['./component.css'],
+})
+export class SpaceAddComponent implements OnInit {
+	createForm: FormGroup;
+	controlAutocomplete = new FormControl('');
+
+	spaces: Observable<Space[]>;
```

Comment:
В ангуларе видимо принято публичные Observable декларировать с долларом на конце.
Восклицательный знак указывает компилятору что эта переменная гарантированно не может быть null или undefined и позволяет убрать лишние проверки на null и undefined. Это было в [туториале](https://angular.io/tutorial/tour-of-heroes/toh-pt6#asyncpipe), давай тоже добавим пожалуйста.
```suggestion
	spaces$!: Observable<Space[]>;
```

Reply:
Готово.